controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/**
+ * @file controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class NavigationMenuForm
+ * @ingroup controllers_grid_navigationMenus_form
+ * @see NavigationMenu
+ *
+ * @brief Form for manager to create/edit NavigationMenus.
+ */
+
+
+import('lib.pkp.classes.form.Form');
+
+class NavigationMenuForm extends Form {
+	/** @var int Context ID */
+	var $_contextId;
+
+	/** @var $navigationMenu NavigationMenu The menu being edited */
+	var $_navigationMenuId;
+
+	/**
+	 * Constructor
+	 * @param $contextId int Context ID
+	 * @param $navigationMenuId int NavigationMenu Id
+	 */
+	function __construct($contextId, $navigationMenuId = null) {
+		$this->_navigationMenuId = !empty($navigationMenuId) ? (int) $navigationMenuId : null;
+		$this->_contextId = $contextId;
+
+		parent::__construct('controllers/grid/navigationMenus/form/navigationMenuForm.tpl');
+
+		$this->addCheck(new FormValidator($this, 'title', 'required', 'manager.announcementTypes.form.typeNameRequired'));
+
+		$this->addCheck(new FormValidatorPost($this));
+		$this->addCheck(new FormValidatorCSRF($this));
+	}
+
+	/**
+	 * Get a list of localized field names for this form
+	 * @return array
+	 */
+	function getLocaleFieldNames() {
+		return null;
+	}
+
+	/**
+	 * @copydoc Form::fetch()
+	 */
+	function fetch($request) {
+		$templateMgr = TemplateManager::getManager($request);
+
+		$themePlugins = PluginRegistry::getPlugins('themes');
+		if (is_null($themePlugins)) {
+			$themePlugins = PluginRegistry::loadCategory('themes', true);
+		}
+		$enabledThemes = array();
+		$activeThemeOptions = array();
+		foreach ($themePlugins as $themePlugin) {
+			$enabledThemes[basename($themePlugin->getPluginPath())] = $themePlugin->getDisplayName();
+			if ($themePlugin->isActive()) {
+				$activeThemeOptions = $themePlugin->getOptionsConfig();
+			}
+		}
+
+		$activeThemeNavigationAreas = array();
+		if (!count($activeThemeOptions['navigationMenuArea']) == 0) {
+			$activeThemeNavigationAreas = array('' => __('common.none'));
+		}
+
+		foreach ($activeThemeOptions['navigationMenuArea'] as $navigationMenuArea) {
+			$activeThemeNavigationAreas[$navigationMenuArea] = $navigationMenuArea;
+		}
+
+		$navigationMenuItemDao = DAORegistry::getDAO('NavigationMenuItemDAO');
+		$navigationMenuItems = $navigationMenuItemDao->getByContextId($request->getContext()->getId())
+				->toArray();
+		$assignedItems = $navigationMenuItemDao->getByMenuId($this->_navigationMenuId)
```

Comment:
Around here is one of the sneaky cases where it might be possible to smuggle in an ID from another journal's menus. The context ID is coming from the `$request` object, but we're trusting the user-supplied `navigationMenuId` without making sure the two match.